### PR TITLE
Progressbar for the console

### PR DESCRIPTION
**Pull request by rpls**
This revamps the entire logging of `mupq`. Pretty much kicks out all the logging from the console, except errors and test pass/fail messages. This frees up the clutter and makes space for a progress bar. The bar is based on the `tqdm` package, which also has support for time tracking etc.

For even longer tests, I also added progress tracking for the longer running tests. Especially the `test` and `speed` test can be quite long-running due to their many iterations, so the gui comes with stacked progress bars (one for the schemes, one for the iterations).

Since a full log is still useful, the full log is dumped into a `mupq.log` file, which is rotated ten times so you don't immediately loose previous runs.

### CODE REVIEW

File: mupq.py [hunk 1]
```diff
@@ -186,33 +228,43 @@ def get_implementations(self, all=False):
...
-            if self.run_test(implementation) == -1:
-                return -1
+            implementations.append(implementation)
+        with tqdm.tqdm(total=len(implementations), desc=self.test_type) as pb:
+            for implementation in implementations:
+                pb.set_postfix_str(f"{implementation}")
+                if self.run_test(implementation) == -1:
+                    pb.write(f"{implementation} FAIL")
+                    return -1
+                pb.write(f"{implementation} DONE")
```
Comment (mkannwischer) [inline mupq.py:253]: I think it is better to output 'SUCCESSFUL' rather than 'DONE'.

File: mupq.py [hunk 2]
```diff
@@ -286,20 +341,15 @@ def hash_output(self, output):
...
-                     f"IMPLEMENTATION_PATH={path}"]
-        if namespace is not None:
-            makeflags.append(f"MUPQ_NAMESPACE={namespace}")
-        makeflags.extend(self.platform_settings.makeflags)
-        makeflags.append(target)
-        return subprocess.check_call(makeflags)
-
+        if self.testvectorhash[implementation.scheme] != checksum:
+            self.log.error("Test %s - %s Failed!", implementation, self.test_type)
+            return -1
```
Comment (mkannwischer) [inline mupq.py:346]: test.py aborts when there is a test failure, while testvectors.py happily continues.
I think it should be consistent. I have a slight preference for the former.

**Comment by mkannwischer** [comment]
Otherwise this looks good to me.
Thanks, @rpls.

File: mupq.py (hunk 1 above)
Comment (rpls) [reply mupq.py:253]: Done

File: mupq.py (hunk 2 above)
Comment (rpls) [reply mupq.py:346]: I matched the behavior between all tests. I also generally overhauled the error handling a bit. And on the `pqm4` side, I let the entry-scripts exit with a return code if the tests fail.

**Pull request merged**